depmapomics/fingerprinting.py

```diff
@@ -0,0 +1,357 @@
+# fingerprinting.py
+
+import pandas as pd
+import numpy as np
+import dalmatian as dm
+from taigapy import TaigaClient
+from genepy.google import gcp
+import asyncio
+from genepy import terra
+from genepy.utils import helper as h
+from depmapomics import tracker
+from depmapomics.config import *
+from depmapomics import terra as myterra
+
+
+def updateLOD(
+    wm, sampleset, taiga_dataset, taiga_filename, working_dir, upload_to_taiga=True
+):
+    # Here we update the fingerprint LOD matrix on taiga with the new fingerprints
+    # Generate matrix with LOD score for new fingerprint vcfs
+    new_lod_list = []
+    sample_batch_pair_df = wm.get_entities("sample_batch_pair")
+    samples_df = sample_batch_pair_df[
+        sample_batch_pair_df.sample_batch_b.apply(
+            lambda x: x["entityName"] == sampleset
+        )
+    ]["cross_checks_out"].tolist()
+    for batch in samples_df:
+        # could be pd concat
+        df = pd.read_csv(batch, sep="\t", comment="#")
+        lod_mat = df.pivot(
+            index="LEFT_SAMPLE", columns="RIGHT_SAMPLE", values="LOD_SCORE"
+        )
+        new_lod_list.append(lod_mat)
+    new_lod_mat = pd.concat(new_lod_list)
+    new_lod_mat.index.name = None
+    new_lod_mat = new_lod_mat.T
+
+    # Update LOD matrix ( have to update (A+a)*(B+b) = (AB)+(aB)+(Ab)+(ab))
+    tc = TaigaClient()
+    prev_lod_mat = tc.get(name=taiga_dataset, file=taiga_filename)
+    new_ids = set(new_lod_mat.index)
+    old_ids = set(prev_lod_mat.index) - set(new_ids)
+    updated_lod_mat = pd.concat(
+        (prev_lod_mat.loc[old_ids, old_ids], new_lod_mat.loc[new_ids, old_ids]), axis=0
+    )
+    updated_lod_mat = pd.concat(
+        (
+            updated_lod_mat.loc[new_ids.union(old_ids), old_ids],
+            new_lod_mat.transpose().loc[new_ids.union(old_ids, new_ids)],
+        ),
+        axis=1,
+    )
+    updated_lod_mat.to_csv(working_dir + taiga_filename + ".csv")
+
+    # Upload updated LOD matrix to Taiga
+    if upload_to_taiga:
```

Comment:
same here. let's keep this in the notebook for now so the function is more generic

Reply:
would it be necessary to add a _CCLEfingerprint wrapper and put all the taiga stuff under it?